File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepagePresenter.java
package com.wayfair.themoviedb.feature.homepage;

import com.pubbix.base.BasePresenter;

import javax.inject.Inject;

public class HomepagePresenter extends BasePresenter<HomepageFragment, HomepageContract.Interactor>
implements HomepageContract.Presenter{
    private HomepageFragment view;
    private HomepageContract.Interactor interactor;
    private HomepageContract.Router router;

    @Inject
    public HomepagePresenter(HomepageFragment view,
                             HomepageContract.Interactor interactor,
                             HomepageContract.Router router) {
        this.view = view;
        this.interactor = interactor;
        this.interactor.setPresenter(this);
        this.router = router;
    }

    @Override
    public void setUpViews() {
        view.setOnQueryTextListener();
    }

    @Override
    public void search(String searchTerms) {
        router.goToSearchPage(searchTerms);
    }

    @Override
    public void setInteractor(HomepageContract.Interactor interactor) {

    }
}

File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepagePresenter.kt
package com.wayfair.themoviedb.feature.homepage

import com.pubbix.base.BasePresenter

import javax.inject.Inject

class HomepagePresenter @Inject
constructor(private val interactor: HomepageContract.Interactor,
            private val router: HomepageContract.Router) : BasePresenter<HomepageContract.View,
        HomepageContract.Interactor>(), HomepageContract.Presenter {

    init {
        this.interactor.setPresenter(this)
    }

    override fun search(searchTerms: String) {
        router.goToSearchPage(searchTerms)
    }
}
